llap-server/src/java/org/apache/hadoop/hive/llap/cache/SerDeLowLevelCacheImpl.java

```diff
@@ -53,6 +61,18 @@
   private final long cleanupInterval;
   private final LlapDaemonCacheMetrics metrics;
 
+  /// Shared singleton MetricsSource instance for all FileData locks
+  private static final MetricsSource LOCK_METRICS;
+
+  static {
```

Comment:
wondering if we can inject the LockMetrics same way we are doing it for `org.apache.hadoop.hive.llap.metrics.LlapDaemonCacheMetrics`. it is Done at `org.apache.hadoop.hive.llap.io.api.impl.LlapIoImpl#LlapIoImpl` ? that way we don't have to add this static block ? what you think ?

Reply:
Good point. I will do that for all the LOCK_METRICS definitions that I added statically. This means, that there are actually three different metrics containers that will be added to LlapIoImpl.